Sources/ConcordiumSwiftSDK/Model/Account.swift

```diff
@@ -16,9 +49,373 @@ struct AccountAddress {
+                                $0.value
+                            }
+                    )
+            )
+        }
+    }
+}
+
+typealias EncryptedAmount<C> = Data // is 'C' in Rust impl but the API returns Data
+
+struct AccountEncryptedAmount {
+    let selfAmount: EncryptedAmount<ArCurve>
+    let startIndex: UInt64
+    let aggregatedAmount: Data?
+    let incomingAmounts: [EncryptedAmount<ArCurve>]
+
+    static func fromGrpcType(_ grpc: Concordium_V2_EncryptedBalance) -> AccountEncryptedAmount {
+        AccountEncryptedAmount(
+                selfAmount: grpc.selfAmount.value,
+                startIndex: grpc.startIndex,
+                aggregatedAmount: grpc.aggregatedAmount.value,
+                incomingAmounts: grpc.incomingAmounts.map {
+                    $0.value
+                }
+        )
+    }
+}
+
+// TODO: Check out Java SDK for this...
+typealias BakerElectionVerifyKey = PublicKey
+typealias BakerSignatureVerifyKey = PublicKey
+typealias BakerAggregationVerifyKey = PublicKey
+
+typealias BakerId = AccountIndex
+
+struct BakerInfo {
+    let bakerId: BakerId
+    let bakerElectionVerifyKey: BakerElectionVerifyKey
+    let bakerSignatureVerifyKey: BakerSignatureVerifyKey
+    let bakerAggregationVerifyKey: BakerAggregationVerifyKey
+
+    static func fromGrpcType(_ grpc: Concordium_V2_BakerInfo) -> BakerInfo {
+        BakerInfo(
+                bakerId: grpc.bakerID.value,
+                bakerElectionVerifyKey: grpc.electionKey.value,
+                bakerSignatureVerifyKey: grpc.signatureKey.value,
+                bakerAggregationVerifyKey: grpc.aggregationKey.value
+        )
+    }
+}
+
+enum StakePendingChange {
+    case reduceStake(newStake: Amount, effectiveTime: Date)
+    case removeStake(effectiveTime: Date)
+
+    static func fromGrpcType(_ grpc: Concordium_V2_StakePendingChange) -> StakePendingChange? {
+        switch grpc.change {
+        case nil:
+            return nil
+        case let .reduce(r):
+            return .reduceStake(newStake: r.newStake.value, effectiveTime: dateFromUnixTimeMillis(r.effectiveTime.value))
+        case let .remove(r):
+            return .removeStake(effectiveTime: dateFromUnixTimeMillis(r.value))
+        }
+    }
+}
+
+struct AmountFraction {
+    let partsPerHundredThousand: UInt32
+
+    static func fromGrpcType(_ grpc: Concordium_V2_AmountFraction) -> AmountFraction {
+        AmountFraction(partsPerHundredThousand: grpc.partsPerHundredThousand)
+    }
+}
+
+enum OpenStatus: Int {
+    case openForAll = 0
+    case closedForNew = 1
+    case closedForAll = 2
+
+    static func fromGrpcType(_ grpc: Concordium_V2_OpenStatus) -> OpenStatus {
+        OpenStatus(rawValue: grpc.rawValue)!
+    }
+}
+
+struct CommissionRates {
+    let finalization: AmountFraction
+    let baking: AmountFraction
+    let transaction: AmountFraction
+
+    static func fromGrpcType(_ grpc: Concordium_V2_CommissionRates) -> CommissionRates {
```

Comment:
The `fromGrpc` method is more or less ubiquitous, so we could consider adding a generic protocol for it. Then again, we would need to add two protocols - one for `fromGrpcType` methods that throws, and one that doesn't, so it MIGHT be overkill for our purposes

Reply:
What would be the advantage of that? Also, can you have static functions in protocols?

Reply:
Ha - I missed the static keyword - which is weird since it's a factory method :D

The only advantage in our case would be to ensure naming consistency.   

Swift protocols can define static members, BUT the static members can't refer to the protocol's associatedtype, so that settles it :)